appveyor_after_build.ps1

```diff
@@ -0,0 +1,358 @@
+    }
+    else
+    {
+        $response = InvokePostRestMethod "/api/profiles/set_as_default?language=cs&name=$testQualityProfileName" $true    
+    }
+}
+
+function VerifyAnalysisResults
+{    
+    WaitForAllBackgroundTasksToComplete
+
+    $expectedRuleViolations = @("fxcop:DoNotPassLiteralsAsLocalizedParameters", "fxcop:DoNotRaiseReservedExceptionTypes", "csharpsquid:S2228", "csharpsquid:S1134")
+
+    $response = InvokeGetRestMethod "/api/issues/search?hideRules=true"
+
+    Assert ($response.paging.total -eq 4) "There should only be 4 issues"
+    Assert ($response.issues.Length -eq 4) "There should only be 4 issues"
+
+    $brokenRules = $response.issues.rule
+    $ruleDiff = Compare-Object $response.issues.rule $expectedRuleViolations
+
+    Assert ($ruleDiff.Length -eq 0) "Expected 4 rules to be broken. Rules found: $brokenRules. Expected: $expectedRuleViolations"
+    
+    foreach ($issue in $issues)
+    {
+        Assert ($issue.project -eq $sqProjectKey) "An issue was found not belonging to the project under test. Instead, it belongs to " + $issue.project        
+    }
+}
+
+function WaitForAllBackgroundTasksToComplete
+{
+    # Async execution happens on SQ 5+ only
+    if (IsVersion5OrGreater)
+    {
+        $tasksInQueue = $true        
+        while ($tasksInQueue)
+        {
+            $response = InvokeGetRestMethod "/api/ce/queue" $true
+            $validResponse = $response.PSobject.Properties.name -match "tasks"
+
+            if (!$validResponse)
+            {
+                throw "The internal API /api/ce/queue did not produce the expected response"
+            }
+
+            if ($response.tasks)
+            {
+                Start-Sleep -s 1
+            }
+            else
+            {
+                $tasksInQueue = $false
+            }
+        }
+    }
+}
+
+function BuildAndAnalyzeProjectUnderTest
+{
+    echo "Step 5.1: Finding msbuild"
+    $msbuildPath = FindLatestMsBuildPath
+    echo "Using the latest msbuild found from $msbuildPath"
+
+    echo "Step 5.2: Locating the MSBuild.SonarQube.Runner.exe"
+    $bootstrapperPath = FindSingleFile ([System.IO.Path]::Combine($env:APPVEYOR_BUILD_FOLDER, "DeploymentArtifacts", "BuildAgentPayload", $env:configuration)) "MSBuild.SonarQube.Runner.exe"
+
+    echo "Step 5.3: Locating the project under test"
+    $projectUnderTestPath = FindSingleFile ([System.IO.Path]::Combine($env:APPVEYOR_BUILD_FOLDER, "IntegrationTestProject")) "ProjectUnderTest.sln"
+
+    echo "Step 5.4: Waiting for SQ to have started"
+    WaitForSonarQubeToHaveStarted
+        
+    echo "Step 5.5: Upload the test quality profile and make it default"
+    UploadTestQualityProfile
+    SetTestQualityProfileAsDefault
+
+    echo "Step 5.6: Begining analysis"
+    RunProcessAndWaitForFinish $bootstrapperPath @("begin", "/k:$sqProjectKey", "/n:$sqProjectKey", "/v:1")
```

Comment:
RunProcessAndWaitForFinish does not have a timeout mechanism. However the appveyor caps test execution at 40 min.